From 1ec7b694b55c70d06f262a96c320709b70d20bcb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tomasz=20Pawe=C5=82=20Gajc?= <tpgxyz@gmail.com>
Date: Mon, 11 May 2020 14:49:06 +0200
Subject: [PATCH] install dependant libs too

By default rng-tools are compiled with pkcs11 support.
Make sure opensc-pkcs11.so library is installed inside initramfs to prevent error on boot
---
 modules.d/06rngd/module-setup.sh | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/modules.d/06rngd/module-setup.sh b/modules.d/06rngd/module-setup.sh
index 354bd0bb..f07a0cb9 100644
--- a/modules.d/06rngd/module-setup.sh
+++ b/modules.d/06rngd/module-setup.sh
@@ -33,6 +33,8 @@ check() {
 install() {
     inst rngd
     inst_simple "${moddir}/rngd.service" "${systemdsystemunitdir}/rngd.service"
+    # make sure dependant libs are installed too
+    inst_libdir_file opensc-pkcs11.so
 
     systemctl -q --root "$initdir" add-wants sysinit.target rngd.service
 }

